Esdk 420 build very first basic app tuto

Tschasmine opened the pull request.

Tschasmine committed:
implement basic customer info displaying

Tschasmine committed:
integration test for displaying vendor info

Tschasmine committed:
implement basic trading partner info displaying

Tschasmine committed:
check zipcode field value in tests

Tschasmine committed:
fill zip code field in infosystem

Tschasmine committed:
zip code selection test

Tschasmine committed:
evaluate zip code for selection

Tschasmine committed:
use real addresses in test data

Tschasmine committed:
latitude and longitude are displayed

Tschasmine committed:
implement displaying of latitude and longitude

Tschasmine committed:
use new gradle plugin version 0.8.0

rhaendel committed:
Fix docker registry URL in README

rhaendel committed:
Add a first Jenkinsfile

rhaendel committed:
The Jenkinsfile want's to know the app's version

rhaendel committed:
Jenkinsfile: Docker login to partner.registry.abas.sh

The intra.registry is the wrong one for pulling partner-Essentials images.

Tschasmine committed:
refactor EventHandler class

rhaendel committed:
Use a Scanner instead of a Reader to get the response text as String

rhaendel committed:
Organize imports and consistent order of 'static final'

Tschasmine committed:
fix package in logging.custom.properties

rhaendel commented on src/main/java/de/abas/app/g30l0/GeoLocationEventHandler.java:
```diff
@@ -0,0 +1,100 @@
+		}
+	}
+
+	private List<TradingPartner> selectBusinessPartners(DbContext ctx, String swd, String zipCode) {
```
Method should be named `selectTradingPartners` to better express what it really does.

Tschasmine committed:
Merge branch 'ESDK-420-build-very-first-basic-app-tuto' of https://github.com/Tschasmine/geoLocation into ESDK-420-build-very-first-basic-app-tuto

rhaendel commented on src/main/java/de/abas/app/g30l0/GeoLocationEventHandler.java:
```diff
@@ -0,0 +1,100 @@
+
+	private List<TradingPartner> selectBusinessPartners(DbContext ctx, String swd, String zipCode) {
+		List<TradingPartner> tradingPartners = new ArrayList<>();
+		tradingPartners.addAll(ctx.createQuery(buildSelection(Customer.class, swd, zipCode)).execute());
```
Just a note, nothing to change here:
Always be careful when collecting database objects in a collection. There can be very many of them and fill the heap space. In production this could lead to a problem and it would be better to use an iterator and fill one infosystem row after another while iterating the number of hits.

rhaendel commented on src/main/java/de/abas/app/g30l0/GeoLocationEventHandler.java:
```diff
@@ -0,0 +1,100 @@
+			row.setTown(tradingPartner.getTown());
+			row.setState(tradingPartner.getStateOfTaxOffice());
+			try {
+				JSONObject place = getPlace(tradingPartner);
```
`location` would be better than `place`.

rhaendel committed:
Make TradingPartnerSelection#buildSelection() a bit more typesafe and specific

rhaendel committed:
Use only one instance of OpenStreetMapGeolocationResolver

and do the web service request in a non-constructor method.

rhaendel committed:
OpenStreetMapGeolocationResolver: location instead of place, use some String constants

rhaendel committed:
ClientSideInfosystemTest: make setup() and cleanup() the same abstraction level

and use QueryUtil.getFirst() to get the one selection result.
iterator().next() only gets the first result and lets the query open if
there is another one. QueryUtil.getFirst() closes the query.
